src/ws2st/refactor.hpp

```diff
@@ -0,0 +1,243 @@
+/**
+ * @file refactor.hpp
+ * @author james
+ * @brief Idea is to write out the results of the refactoring using the classes defined here
+ * @version 0.1
+ * @date 2024-01-22
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#pragma once
+
+#include "common.hpp"
+
+namespace wash {
+
+    std::optional<std::string> getSourceText(ASTContext* ctx, SourceRange srcRange);
+
+    namespace refactoring {
+
+        enum TypeKind { SCALAR, VECTOR };
```

Comment:
I think a scoped enum (`enum class`) would be a safer bet here. Unscoped enums are only useful if you wanted `TypeKind` to implictly convert to its underlying type (an `int`), which I don't think is the case here.